monitor.go

```diff
@@ -22,34 +27,66 @@ type Exit struct {
 // These methods should match the methods exposed by exec.Cmd to provide
 // a consistent experience for the caller
 type ProcessMonitor interface {
-	Start(*exec.Cmd) (chan Exit, error)
+	Start(context.Context, *exec.Cmd) (chan Exit, error)
 	Wait(*exec.Cmd, chan Exit) (int, error)
 }
 
+func DefaultMonitor(defaultSignal os.Signal, killTimeout time.Duration) ProcessMonitor {
+	return &defaultMonitor{
+		defaultSignal: defaultSignal,
+		killTimeout:   killTimeout,
+	}
+}
+
 type defaultMonitor struct {
+	defaultSignal os.Signal
+	killTimeout   time.Duration
 }
 
-func (m *defaultMonitor) Start(c *exec.Cmd) (chan Exit, error) {
+func (m *defaultMonitor) Start(ctx context.Context, c *exec.Cmd) (chan Exit, error) {
 	if err := c.Start(); err != nil {
 		return nil, err
 	}
 	ec := make(chan Exit, 1)
+	waitDone := make(chan struct{}, 1)
+	go func() {
+		select {
+		case <-ctx.Done():
+			if m.defaultSignal == nil {
+				c.Process.Signal(unix.SIGKILL)
+			} else {
+				c.Process.Signal(m.defaultSignal)
+				if m.killTimeout > 0 {
```

Comment:
If there's no killtimeout set better use a default value, the process may never die otherwise.